source/code/shared/tools/scx_ssl_config/scxsslcert.cpp

```diff
@@ -342,9 +347,13 @@ void SCXSSLCertificate::DoGenerate()
         string outfile(StrToMultibyte(m_CertPath));
         string keyout(StrToMultibyte(m_KeyPath));
 
-        ManagedResource res1(ERR_load_crypto_strings,        ERR_free_strings);
-        ManagedResource res2(SSL_OpenSSL_add_all_algorithms, EVP_cleanup);
-        ManagedResource res3(ENGINE_load_builtin_engines,    ENGINE_cleanup);
+        #if OPENSSL_VERSION_NUMBER <= 0x100fffffL // SSL 1.0.x or lower?
+            ManagedResource res1(ERR_load_crypto_strings,        ERR_free_strings);
+            ManagedResource res2(SSL_OpenSSL_add_all_algorithms, EVP_cleanup);
+            ManagedResource res3(ENGINE_load_builtin_engines,    ENGINE_cleanup);
+        #else
+            ManagedResource res3(ENGINE_load_builtin_engines,    NoOp_Destructor);
```

Comment:
So you sure you don't need to add algorithms and crypto strings? (I'm honestly not sure without looking more closely, so I'm actually askin')

Also, if at 355 you only have one resource, mind as well call it `res1`. But verify that you only need one resource, particularly if the customer overrides the specified crypto strings via the OMI configuraiton file.

Reply:
As of version 1.1.0 OpenSSL will automatically allocate all resources that it needs so no explicit initialisation is required. Similarly it will also automatically deinitialise as required. 

I will rename res3  as res1 at 355.

Reply:
Fixed